esp-link/cgitelnet.c

```diff
@@ -0,0 +1,79 @@
+#include <esp8266.h>
+#include "cgi.h"
+#include "config.h"
+#include "serbridge.h"
+
+// Cgi to return choice of Telnet ports
+int ICACHE_FLASH_ATTR cgiTelnetGet(HttpdConnData *connData) {
+  if (connData->conn==NULL) return HTTPD_CGI_DONE; // Connection aborted
+
+  char buff[1024];
+  int len;
+
+  len = os_sprintf(buff,
+      "{ \"telnet-port1\":%d, \"telnet-port2\":%d }",
+      flashConfig.telnet_port1, flashConfig.telnet_port2);
+
+  jsonHeader(connData, 200);
+  httpdSend(connData, buff, len);
+  return HTTPD_CGI_DONE;
+}
+
+// Cgi to change choice of Telnet ports
+int ICACHE_FLASH_ATTR cgiTelnetSet(HttpdConnData *connData) {
+  if (connData->conn==NULL) {
+    return HTTPD_CGI_DONE; // Connection aborted
+  }
+
+  int8_t ok = 0;
+  uint8_t port1, port2;
```

Comment:
Of course! I got the thought of a short int = 8bit stuck in my head yesterday.
Once I realized that... unsigned 16bit should get me there.